src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -991,6 +1007,12 @@
   ; gone through this function
   (map #(get-in query-rec [:projections % :field]) (sort columns)))
 
+(defn strip-function-calls
+  [column]
+  (let [split (group-by #(= "function" (first %)) column)]
```

Comment:
no, because the order isn't guaranteed if true and false are not both represented.